.github/workflows/docker-build-only.yml

```diff
@@ -0,0 +1,84 @@
+# This workflow builds the server docker image and exposes it as an artifact which may be used by
+# other GitHub workflow jobs. This is useful for creating docker images that you want to use to
+# test a certain commit of server without publishing them.
+
+name: Build Docker Image as Artifact
+
+on:
+  workflow_call:
+    inputs:
+      # TODO: Shouldn't be required. If not set, use existing submodule.
+      temporal-server-repo-ref:
+        type: string
+        required: true
+      temporal-server-repo-path:
+        type: string
+        default: "temporalio/temporal"
+      docker-builds-repo-ref:
+        type: string
+        default: "main"
+
+jobs:
+  build-image:
+    # TODO: use bigger runner when available. Seems like they're stuck or not enough?
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout docker builds repository
+        uses: actions/checkout@v3
+        with:
+          # Must specify repo path, or it will use path of calling workflow
+          repository: temporalio/docker-builds
+          ref: ${{ inputs.docker-builds-repo-ref }}
+          submodules: "true"
+
+      - name: Checkout temporal server repository
```

Comment:
instead of doing this, would it be possible to just go into the `temporal` submodule and fetch+checkout a new ref? it seems like it'd be more efficient (since most of it is already there) and then you don't need to mess with the directory name

Reply:
Yeah, we could do that.